data_preprocessing.py

```
@@ -0,0 +1,45 @@
+        return len(urlparse(url).path.split('/')[1])
+    except:
+        return 0
+df['fd_length'] = df['url'].apply(fd_length)
+
+df['count-'] = df['url'].apply(lambda x: x.count('-'))
+df['count.'] = df['url'].apply(lambda x: x.count('.'))
+
+def get_tld_length(url):
+    try:
+        return len(get_tld(url, fail_silently=True))
+    except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)